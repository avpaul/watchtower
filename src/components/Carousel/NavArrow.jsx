import React from 'react';
import PropTypes from 'prop-types';

const NavArrow = props => {
  const { buttonClass, iconClass, onClick } = props;
  return (
    <div
      role="presentation"
      className={`slick-arrow ${buttonClass}`}
      onClick={onClick}
    >
      <span role="presentation" className={`fa ${iconClass}`} />
    </div>
  );
};

NavArrow.propTypes = {
  buttonClass: PropTypes.string.isRequired,
  iconClass: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default NavArrow;
