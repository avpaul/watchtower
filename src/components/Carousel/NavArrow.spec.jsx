import React from 'react';
import { mount } from 'enzyme';
import NavArrow from './NavArrow';

describe('NavArrow Component Test Suite', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<NavArrow />);
  });

  it('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
