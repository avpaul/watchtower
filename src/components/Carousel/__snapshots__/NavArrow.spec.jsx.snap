// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavArrow Component Test Suite matches snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <NavArrow />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div
        role="presentation"
        class="slick-arrow undefined"
>
        <span
                role="presentation"
                class="fa undefined"
        />
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="fa undefined"
          role="presentation"
/>,
        "className": "slick-arrow undefined",
        "onClick": undefined,
        "role": "presentation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <span
            role="presentation"
            class="fa undefined"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa undefined",
            "role": "presentation",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div
          role="presentation"
          class="slick-arrow undefined"
>
          <span
                    role="presentation"
                    class="fa undefined"
          />
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="fa undefined"
            role="presentation"
/>,
          "className": "slick-arrow undefined",
          "onClick": undefined,
          "role": "presentation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <span
              role="presentation"
              class="fa undefined"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa undefined",
              "role": "presentation",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
